core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java

```diff
@@ -180,25 +147,29 @@ private static PageId getPageIdFromKey(byte[] key) {
   }
 
   @Override
-  public long pages() {
-    return mSize.get();
-  }
-
-  @Override
-  public long bytes() {
-    return mBytes.get();
-  }
-
-  @Override
-  public Collection<PageInfo> getPages() {
-    try (RocksIterator iter = mDb.newIterator()) {
-      return Streams.stream(new PageIterator(iter)).collect(Collectors.toList());
+  public Stream<PageInfo> getPages() throws IOException {
+    try {
+      byte[] confData = mDb.get(CONF_KEY);
+      Cache.PRocksPageStoreOptions pOptions = mOptions.toProto();
+      if (confData != null) {
+        Cache.PRocksPageStoreOptions persistedOptions =
+            Cache.PRocksPageStoreOptions.parseFrom(confData);
+        if (!persistedOptions.equals(pOptions)) {
+          mDb.close();
+          throw new IOException("Inconsistent configuration for RocksPageStore");
+        }
+      }
+      mDb.put(CONF_KEY, pOptions.toByteArray());
+      return Streams.stream(new PageIterator(mDb.newIterator()));
```

Comment:
What happens when we call close on this stream, is it a no-op?

Reply:
correct

Reply:
fixed by adding closer to `onClose`